src/index.html

```diff
@@ -11,7 +11,216 @@
     <link rel="stylesheet" href="./style.css">
   </head>
   <body>
-    <h1>HTML Form</h1>
+    <form name="form">
+      <fieldset class="perInfo" name="Personal">
+        <legend>Personal information</legend>
+        
+        <div class="per">
+          <label for="surName">
+            Surname:
+          </label>
+          <input 
+            type="text"
+            name="username" 
+            required
+          >
+        </div>
+
+        <div class="per">
+          <label for="name">
+            Name:
+          </label>
+          <input 
+            id="name"
+            type="text"
+            name="name" 
+            required
+          >
+        </div>
+
+        <div class="per">
+          <label for="age">
+            How old are You?
+          </label>
+          <input 
+            id="age" 
+            class="age"
+            type="number" 
+            name="age" 
+            value="12"
+            min="1"
+            max="100"
+          >
+        </div>
+
+        <div class="per">
+          <label for="date">
+            Full date of birth:
+          </label>
+          <input 
+            id="date" 
+            type="date"
+            name="date" 
+            required
+          >
+        </div>
+
+        <div class="per">
+          <label for="checkbox">
+            I accept the term of the agreement
+          </label>
+          <input 
+            id="checkbox" 
+            type="checkbox" 
+            name="agreement" 
+            required
+          >
+        </div>
+
+      </fieldset>
+      <fieldset class="regist">
+        <legend>Registration</legend>
+
+        <div class="reg">
+          <label for="email">
+            E-mail:
+          </label>
+          <input 
+            id="email" 
+            type="email" 
+            name="email" 
+            placeholder="email@example.com" 
+            required 
+            multiple
+          >
+        </div>
+
+        <div class="reg">
+          <label for="password">
+            Password:
+          </label>
+          <input 
+            id="password" 
+            type="password" 
+            name="password" 
+            minlength="10" 
+            maxlength="20" 
+            required
+          >
+        </div>
+
+      </fieldset>
+      <fieldset class="fact">
+        <legend>An interesting fact about you!</legend>
+
+        <div class="facting">
+          <label for="radio">
+            Do you love cats?
+          </label>
+          <input 
+            id="radio" 
+            type="radio" 
+            name="radio" 
+            value="y"
+          > Yes
+          <input 
+            id="radio1" 
+            type="radio" 
+            name="radio" 
+            value="n"
+          > No
+        </div>
```

Comment:
here is different, put each input into its own label Yes/No, and Do you love cats? should not be a label, just leave in in div